packages/lib-classifier/src/components/Classifier/components/ImageToolbar/components/ZoomInButton/ZoomInButtonContainer.js

```diff
@@ -17,10 +17,31 @@ function storeMapper (stores) {
 @inject(storeMapper)
 @observer
 class ZoomInButtonContainer extends React.Component {
-  render () {
+  constructor () {
+    super()
+    this.onPointerDown = this.onPointerDown.bind(this)
+    this.onPointerUp = this.onPointerUp.bind(this)
+    this.timer = ''
+  }
+
+  onPointerDown (event) {
     const { zoomIn } = this.props
+    this.timer = setInterval(zoomIn, 100)
+  }
+
+  onPointerUp (event) {
+    clearInterval(this.timer)
```

Comment:
Yoiks, this is a bit problematic - if I click down on the Zoom In button, then move the mouse cursor out of the button BEFORE lifting the mouse click, then the zoom in action will trigger _ad infinitum._

If we're using the `intervals/timer` solution for the Zoom In Button (I'm not sure if this comment on Slack - _"I figured out how to get a Grommet Button to fire continuously on long press: wrap it in a container that listens to pointerdown"_ - meant that you already have a different solution in mind) - then the `clearInterval()` also needs to trigger at  `onPointerOut`, `onPointerLeave`, and/or `onPointerCancel`.

Also, if you want to be as paranoid as me, it's good to call `clearInterval(this.timer)` before `this.timer = setInterval(zoomIn, 100)` in `onPointerDown`, so it's impossible to create multiple active timers.

Reply:
That's a good point about starting multiple timers. I've added pointer capture to solve the problem of releasing the pointer outside the button.